analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant.cs

```diff
@@ -82,24 +81,17 @@ private void CheckInstanceMembers(SonarSyntaxNodeReportingContext c, TypeDeclara
                 return;
             }
 
-            var initializerDeclarations = GetInitializerDeclarations<ConstructorDeclarationSyntax>(c, typeInitializers);
-            foreach (var memberSymbol in initializedMembers.Keys)
+            var constructorDeclarations = GetConstructorDeclarations<ConstructorDeclarationSyntax>(c, constructors);
+            foreach (var kvp in initializedMembers)
             {
                 // the instance member should be initialized in ALL instance constructors
                 // otherwise, initializing it inline makes sense and the rule should not report
-                if (initializerDeclarations.All(constructor =>
-                    {
-                        if (constructor.Node.Initializer != null
-                            && constructor.Node.Initializer.ThisOrBaseKeyword.IsKind(SyntaxKind.ThisKeyword))
-                        {
-                            // Calls another ctor, which is also checked.
-                            return true;
-                        }
-
-                        return IsSymbolFirstSetInCfg(memberSymbol, constructor.Node, constructor.Model, c.Cancel);
-                    }))
+                if (constructorDeclarations.TrueForAll(constructor =>
+                    // Calls another ctor, which is also checked:
+                    constructor is { Node.Initializer.ThisOrBaseKeyword.RawKind: (int)SyntaxKind.ThisKeyword }
+                    || IsSymbolFirstSetInCfg(kvp.Key, constructor.Node, constructor.Model, c.Cancel)))
```

Comment:
`constructor` should be named `x` as it is in a lambda (see [here](https://github.com/SonarSource/sonar-dotnet/blob/master/docs/coding-style.md#naming-conventions)).

```suggestion
                if (constructorDeclarations.TrueForAll(x =>
                    // Calls another ctor, which is also checked:
                    x is { Node.Initializer.ThisOrBaseKeyword.RawKind: (int)SyntaxKind.ThisKeyword }
                    || IsSymbolFirstSetInCfg(kvp.Key, x.Node, x.Model, c.Cancel)))
```